hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -594,7 +599,7 @@ class HAZELCAST_API ClientMessage
         std::is_same<std::pair<typename T::value_type::first_type,
                                typename T::value_type::second_type>,
                      typename T::value_type>::value &&
-        std::is_trivial<typename T::value_type::first_type>::value &&
+        hazelcast::util::is_trivial_or_uuid<typename T::value_type::first_type>::value &&
         std::is_trivial<typename T::value_type::second_type>::value,
```

Comment:
```suggestion
        hazelcast::util::is_trivial_or_uuid<typename T::value_type::second_type>::value,
```
there are 6 places in ClientMessage.h which uses `std::is_trivial` and all needs to change right?

Reply:
I have to admit I don't really know what this code does; I just checked where they presumably were supposed to go before.

A use of `is_trivial` only needs to change if you expect that thing to accept `boost::uuid` in that place.

I think I got all the `get` overloads, or did I miss one?

Reply:
@BillyONeal Just sent an update commit for the missing parts. Can you review them if you see any problems. And then, we can merge if the PR is green. It would still be good to see if this code would work in latest boost versions.